### PR TITLE
Mul for MatQ and MatZ

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR implements multiplication for MatQ and MatZ.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/rational/mat_q/arithmetic/mul.rs [hunk 1]
```diff
@@ -50,6 +52,54 @@ impl Mul for &MatQ {
...
+    /// - Panics if the dimensions of `self` and `other` do not match for multiplication.
+    fn mul(self, other: &MatZ) -> Self::Output {
+        if self.get_num_columns() != other.get_num_rows() {
+            panic!("Tried to multiply matrices with mismatching matrix dimensions.");
+        }
+
+        let other_matq = MatQ::from(other);
+        let mut new = MatQ::new(self.get_num_rows(), other.get_num_columns()).unwrap();
+        unsafe { fmpq_mat_mul(&mut new.matrix, &self.matrix, &other_matq.matrix) };
+        new
```
Comment (Marvin-Beckmann) [inline src/rational/mat_q/arithmetic/mul.rs:92]: Please use `fmpq_mat_mul_fmpz_mat`, as we do not have to clone the entire matrix this way.
```suggestion
        let mut new = MatQ::new(self.get_num_rows(), other.get_num_columns()).unwrap();
        unsafe { fmpq_mat_mul_fmpz_mat(&mut new.matrix, &self.matrix, &other.matrix) };
        new
```

File: src/rational/mat_q/arithmetic/mul.rs [hunk 2]
```diff
@@ -50,6 +52,54 @@ impl Mul for &MatQ {
...
+    ///
+    /// let c = &a * &b;
+    /// let d = a * b;
+    /// let e = &c * d;
+    /// let f = c * &e;
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Panics if the dimensions of `self` and `other` do not match for multiplication.
+    fn mul(self, other: &MatZ) -> Self::Output {
```
Comment (Marvin-Beckmann) [inline src/rational/mat_q/arithmetic/mul.rs:84]: see other PR, mul_safe?

**Review by Marvin-Beckmann — CHANGES_REQUESTED** [review]
A test fails and I proposed some small changs as in the other PR

File: src/rational/mat_q/arithmetic/mul.rs (hunk 1 above)
Comment (Marcel583) [reply src/rational/mat_q/arithmetic/mul.rs:92]: I did not know this exists...

File: src/rational/mat_q/arithmetic/mul.rs (hunk 2 above)
Comment (Marcel583) [reply src/rational/mat_q/arithmetic/mul.rs:84]: see other PR

**Review by Marvin-Beckmann — APPROVED** [review]
